test/test_um2netcdf.py

```diff
@@ -651,6 +663,355 @@ def test_fix_units_do_nothing_no_um_units(ua_plev_cube):
+    # lon_v grid from CM2 (which uses the End Game grid).
+    # This grid is offset half a grid cell compared to the standard
+    # New Dynamics longitude grid.
+    lon_u_points = np.arange(0, 360, D_LON_N96, dtype="float32")
+    return lon_u_points, um2nc.LONGITUDE
+
+
+@pytest.fixture
+@to_iris_dimcoord
+def lat_standard_nd_coord():
+    # iris DimCoord imitating the standard latitude
+    # grid from ESM1.5 (which uses the New Dynamics grid).
+    lat_points = np.arange(-90, 91, D_LAT_N96, dtype="float32")
+    return lat_points, um2nc.LATITUDE
+
+
+@pytest.fixture
+@to_iris_dimcoord
+def lon_standard_nd_coord():
+    # iris DimCoord imitating the standard longitude
+    # grid from ESM1.5 (which uses the New Dynamics grid).
+    lon_points = np.arange(0, 360, D_LON_N96, dtype="float32")
+    return lon_points, um2nc.LONGITUDE
+
+
+@pytest.fixture
+@to_iris_dimcoord
+def lat_standard_eg_coord():
+    # iris DimCoord imitating the standard latitude
+    # grid from CM2 (which uses the End Game grid).
+    lat_points = np.arange(-90 + 0.5*D_LAT_N96, 90.,
+                           D_LAT_N96, dtype="float32")
+    return lat_points, um2nc.LATITUDE
+
+
+@pytest.fixture
+@to_iris_dimcoord
+def lon_standard_eg_coord():
+    # iris DimCoord imitating the standard longitude
+    # grid from CM2 (which uses the End Game grid).
+    lon_points = np.arange(0.5*D_LON_N96, 360, D_LON_N96, dtype="float32")
+    return lon_points, um2nc.LONGITUDE
+
+
+class DummyCubeWithCoords(DummyCube):
+    # DummyCube with coordinates, which can be filled with
+    # iris.DimCoord objects for testing.
+    def __init__(self, dummy_cube, coords=None):
+        super().__init__(dummy_cube.item_code,
+                         dummy_cube.var_name,
+                         dummy_cube.attributes,
+                         dummy_cube.units)
+
+        # Set up coordinate dictionary to hold each dummy coordinate, with keys
+        # given by the coordinate names. This ensures that the key used to
+        # access a coordinate via the coord method matches the
+        # coordinate's name.
+        if coords is not None:
+            self.coordinate_dict = {
+                coord.name(): coord for coord in coords
+            }
+        else:
+            self.coordinate_dict = {}
+
+    def coord(self, coordinate_name):
+        return self.coordinate_dict[coordinate_name]
+
+
+def assert_coordinates_are_unmodified(lat_coord, lon_coord):
+    """
+    Helper function to check that a coordinate's attributes match
+    those expected for a coordinate that has not yet been modified
+    by fix_latlon_coords.
+    """
+    for coord in [lat_coord, lon_coord]:
+        assert coord.points.dtype == np.dtype("float32")
+        assert not coord.has_bounds()
+        assert coord.var_name is None
+
+
+def assert_dtype_float64(lat_coord, lon_coord):
+    assert lat_coord.points.dtype == np.dtype("float64")
+    assert lon_coord.points.dtype == np.dtype("float64")
```

Comment:
I thought of a stylistic simplification like this:
```suggestion
def is_float64(lat_coord, lon_coord):  # becomes an is_something() bool function
    return (lat_coord.points.dtype == np.dtype("float64") and
            lon_coord.points.dtype == np.dtype("float64"))
```
Then the `assert` can be done explicitly lower down:
```python
assert is_float64(cube_lat_coord, cube_lon_coord)
```
This has the effect of ensuring the assertion blocks uniformly use `assert` as a keyword.

Reply:
I've added this in 6a4c876